cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1511,6 +1521,13 @@ def _add_compute_resource_launch_template(
                                 ).lower(),
                                 "HeadNodePrivateIp": self._head_eni.attr_primary_private_ip_address,
                                 "DirectoryServiceEnabled": str(self._config.directory_service is not None).lower(),
+                                "Timeout": (
+                                    str(get_attr(self._config, "dev_settings.timeouts.compute_node_bootstrap_timeout"))
+                                    if get_attr(self._config, "dev_settings.timeouts.compute_node_bootstrap_timeout")
+                                    else str(NODE_BOOTSTRAP_TIMEOUT)
+                                )
+                                if self._config.scheduling.scheduler != "slurm"
```

Comment:
I don't understand why we need this condition. I thought that this was valid for both slurm and plugin

Reply:
Timeout here is passed into compute node user_data. For slur, we pass the timeout to slurm configuration and clustermgtd configuration. 
According to the design doc to leave the logic unaltered:
```
For our first-class Slurm integration we will keep propagating the ComputeNodeBootstrapTimeout to the Slurm and clustermgtd settings in order to leave the logic that restores the scheduler nodes status unaltered.
```
I didn't pass the timeout for Slurm to compute node user data

Reply:
The Timeout here is for compute node user_data:
We don’t pass it for slurm because in slurm we want to rely on slurm configuration instead of user_data Timeout.

For passing `compute_node_bootstrap_timeout` to cookbook, we already do it for all scheduler, it is in headnode dna.json.

Reply:
Added in compute node bootstrap timeout